src/ert/_c_wrappers/enkf/lark_parser.py

```diff
@@ -45,6 +46,25 @@ def __str__(self):
         return self.value
 
 
+def assert_is_token(
+    val: Any, kw: Optional[FileContextToken] = None
+) -> TypeGuard[FileContextToken]:
+    if not isinstance(val, FileContextToken):
+        # Should be impossible to reach this through the parser
+        raise ConfigValidationError(
+            f"Unexpected value in keyword: {kw!r}, value: {val!r}",
+            config_file=kw.filename if kw else None,
+        )
+    return True
+
+
+def assert_is_tree(val: Any, filename) -> TypeGuard[FileContextToken]:
+    if not isinstance(val, Tree):
+        # Should be impossible to reach this through the parser
```

Comment:
Maybe make this a docstring to the assert instead:
```python
def assert_is_token(val, filename):
    """Raises an error if val is not of type FileContextToken.
    used to indicate ensure that a value fetched from the ParseTree is actually a token,
   and not Tree. This should be guaranteed by the grammar, but such a case is indicated by 
   this function for two reasons:
   1. as a type guard
   2. Just in case the grammar is more permissive than expected. Then the user still gets some reasonable
       error message.
    """